From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammed Efe Cetin <efectn@protonmail.com>
Date: Thu, 11 Sep 2025 02:47:34 +0300
Subject: Revert "pinctrl: devicetree: do not goto err when probing hogs in
 pinctrl_dt_to_map"

This reverts commit 5b1b4cb46d9514bec053b392d2266f00e0d6ee19.
---
 drivers/pinctrl/devicetree.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/pinctrl/devicetree.c b/drivers/pinctrl/devicetree.c
index 111111111111..222222222222 100644
--- a/drivers/pinctrl/devicetree.c
+++ b/drivers/pinctrl/devicetree.c
@@ -143,14 +143,10 @@ static int dt_to_map_one_config(struct pinctrl *p,
 		pctldev = get_pinctrl_dev_from_of_node(np_pctldev);
 		if (pctldev)
 			break;
-		/*
-		 * Do not defer probing of hogs (circular loop)
-		 *
-		 * Return 1 to let the caller catch the case.
-		 */
+		/* Do not defer probing of hogs (circular loop) */
 		if (np_pctldev == p->dev->of_node) {
 			of_node_put(np_pctldev);
-			return 1;
+			return -ENODEV;
 		}
 	}
 	of_node_put(np_pctldev);
@@ -269,8 +265,6 @@ int pinctrl_dt_to_map(struct pinctrl *p, struct pinctrl_dev *pctldev)
 			ret = dt_to_map_one_config(p, pctldev, statename,
 						   np_config);
 			of_node_put(np_config);
-			if (ret == 1)
-				continue;
 			if (ret < 0)
 				goto err;
 		}
-- 
Armbian

